domain/topn_slow_query.go

```diff
@@ -59,28 +62,87 @@ func (h *slowQueryHeap) Refresh(now time.Time, recent time.Duration) {
 	heap.Init(h)
 }
 
+func (h *slowQueryHeap) Query(count int) []*SlowQueryInfo {
+	save := make([]*SlowQueryInfo, len(h.data))
+	copy(save, h.data)
+
+	sort.Sort(h)
+
+	ret := make([]*SlowQueryInfo, 0, count)
+	for i := len(h.data) - 1; i >= 0 && len(ret) <= count; i-- {
+		ret = append(ret, h.data[i])
+	}
+
+	// Sort breaks the data, recover it here.
```

Comment:
The sorted array still maintains the heap property, we don't need to copy and recover.

Reply:
There are two problems here:
1. To keep top N slow queries, we maintains a min heap, but sort will use a reverse order
2. There is a `count` argument, so the `sorted[count]` is not the same with `heap[len]`